Dockerfile

```diff
@@ -1,29 +1,30 @@
 FROM node:12.18.3-alpine3.12 AS build
-
 WORKDIR /opt/myapp
-
 COPY package*.json ./
-
 RUN yarn install --production
-
 COPY . .
-
 RUN yarn run confd:prod
-
 RUN yarn build
 
-FROM nginx:1.19.1-alpine
 
+FROM nginx:1.19.1-alpine
+RUN addgroup -S app && adduser -S app -G app
 # Install Node for running confd
 RUN set -eux & apk add --no-cache nodejs
-
 COPY entrypoint.sh /entrypoint.sh
-
 WORKDIR /usr/share/nginx/html
-
 COPY --from=build /opt/myapp/build ./
-
 # Move env-config.js to location in which confd script expects it to be located
 RUN mkdir public && cp env-config.js public/env-config.js
 
+#chage nginx config to work without root
+RUN sed -i 's/listen       80;/listen       8080;/g' /etc/nginx/conf.d/default.conf  && \
```

Comment:
maybe create a nginx:1.19.1-alpine base package with this default.conf for use in all other clients

Reply:
we got only 2 nginx services and no registry ATM so I don't think it is relevant.
also if you want to add the permissions to there is can either be unclear to use or problematic to use due to lucking permissions